go/vt/sqlparser/sql.y

```diff
@@ -6229,7 +6229,7 @@ inner_join:
 straight_join:
   STRAIGHT_JOIN
   {
-    $$ = StraightJoinStr
+    $$ = JoinStr
```

Comment:
could we do the conversion to INNER at the GMS layer? I know it's slightly more annoying, but in case we need to actually support the partial ordering at some point

Reply:
It makes sense to do it in GMS instead.
In that case, there don't need to be any changes to vitess.

GMS PR: https://github.com/dolthub/go-mysql-server/pull/2378